### PR TITLE
Improve documentation and location of iaf_chxk_2008 model

### PR DESCRIPTION
**Pull request by heplesser**
This PR clarifies some peculiarities of the iaf_chxk_2008 model by 
- revising the user documentation
- refactoring AHP-related code in the `update()` method
- moving the model to the `precise` module, where it belongs since it emits precise spike times determined by linear interpolation, even though it ignores precise times on input.

### CODE REVIEW

File: nestkernel/node.h [hunk 1]
```diff
@@ -152,14 +152,12 @@ class Node
   virtual bool one_node_per_process() const;
 
   /**
-   * Returns true if the node if it sends/receives -grid events This is
-   * used to discriminate between different types of nodes, when adding
+   * Returns true if the node if it sends/receives off-grid events This is
```
Comment (clinssen) [inline nestkernel/node.h]:
```suggestion
   * Returns true if the node sends/receives off-grid events. This is
```

File: precise/iaf_chxk_2008.h [hunk 2]
```diff
@@ -67,11 +67,13 @@ Name: iaf_chxk_2008 - Conductance based leaky integrate-and-fire neuron model
 Description:
 
 iaf_chxk_2008 is an implementation of a spiking neuron using IAF dynamics with
-conductance-based synapses [1]. It is modeled after iaf_cond_alpha with the
-addition of after hyper-polarization current instead of a membrane potential
-reset. Incoming spike events induce a post-synaptic change of conductance
-modeled by an alpha function. The alpha function is normalized such that an
-event of weight 1.0 results in a peak current of 1 nS at \f$ t = tau_{syn} \f$.
+conductance-based synapses [1]. A spike is emitted when the membrane potential
+is crossed from below. After a spike, and afterhyperpolarizing (AHP) conductance
```
Comment (clinssen) [inline precise/iaf_chxk_2008.h]:
```suggestion
is crossed from below. After a spike, a afterhyperpolarizing (AHP) conductance
```

File: nestkernel/node.h [hunk 3]
```diff
@@ -152,14 +152,12 @@ class Node
...
+   * used to discriminate between different types of nodes when adding
```
[the remark this answers is not in the document]
Comment (heplesser) [reply nestkernel/node.h]: @sarakonradi Thanks, fixed by accepting @clinssen's suggestion.

File: precise/iaf_chxk_2008.h [hunk 4]
```diff
@@ -67,11 +67,13 @@ Name: iaf_chxk_2008 - Conductance based leaky integrate-and-fire neuron model
...
+is activated which repolarizes the neuron over time. Membrane potential is not
```
[the remark this answers is not in the document]
Comment (heplesser) [reply precise/iaf_chxk_2008.h]: Thanks, fixed.

File: precise/iaf_chxk_2008.h [hunk 5]
```diff
@@ -97,6 +99,15 @@ The following parameters can be set in the status dictionary.
...
 \endverbatim
 
+Remarks:
+- In accordance with the original Fortran implementation of the model used in [1], the activation time point for the AHP
+following a spike is detemined by linear interpolation within the time step during which the threshold was crossed.
+- iaf_chxk_2008 neurons therefore emit spikes with precise spike time information, but they ignore precise spike times
+when handling synaptic input.
+- In the original Fortran implementation underlying [1], all previous AHP activation was discarded when a new spike
+occured, leading to reduced AHP currents in particular during periods of high spiking activity. Set `ahp_bug` to `true`
+to obtain this behavior in the model.
```
[the remark this answers is not in the document]
Comment (heplesser) [reply precise/iaf_chxk_2008.h]: I have fixed the formatting now and a few typo's along the way.

**Comment by heplesser** [comment]
@clinssen @sarakonradi I have accepted/addressed your suggestions and hope all is well now.

**Comment by heplesser** [comment]
@sarakonradi I have fixed merge conflicts in this branch. Could you check that I addressed all concerns you raised earlier?

File: precise/iaf_chxk_2008.cpp [hunk 6]
```diff
@@ -410,35 +409,36 @@ nest::iaf_chxk_2008::update( Time const& origin, const long from, const long to
...
-
-      if ( P_.ahp_bug == true )
+      if ( P_.ahp_bug )
       {
-        // Bug in original code ignores AHP conductance from previous spikes
-        S_.y[ State_::G_AHP ] = delta_g_ahp;
-        S_.y[ State_::DG_AHP ] = delta_dg_ahp;
+        // Bug in original Fortran implementation by Casti et al ignores
+        // AHP conductance from previous spikes: at any time, only the
+        // alpha function "kicked off" by the most recent spike is included
```
Comment (sarakonradi) [inline precise/iaf_chxk_2008.cpp]: Could we change `et al` to `et al.`? This is the style we use for in-text citations throughout our documentation.

Comment (heplesser) [reply precise/iaf_chxk_2008.cpp]: Done!

**Comment by heplesser** [comment]
Do not merge this branch at present! This branch shows strange failures on Travis under certain circumstances, which seem unrelated to the changes in the branch. I am currently investigating, so do not merge!

**Pull request merged**